addons/knowledge/models/knowledge_article.py

```diff
@@ -257,10 +257,10 @@ def _compute_user_has_access(self):
         member_permissions = self._get_partner_member_permissions(partner_id.id, article_ids=self.ids)
         for article in self:
             if self.env.user.share:
-                article.user_has_access = member_permissions.get(article.id, "none") != "none"
+                article.user_has_access = member_permissions.get(article.ids[0], "none") != "none"
```

Comment:
add note in docstring to explain that we use ids[0] instead of id to avoid issues with `<newId>` records.